service/src/main/java/uk/nhs/adaptors/gp2gp/gpc/GpcWebClientFilter.java

```diff
@@ -0,0 +1,24 @@
+package uk.nhs.adaptors.gp2gp.gpc;
+
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.ExchangeFilterFunction;
+
+import lombok.extern.slf4j.Slf4j;
+import reactor.core.publisher.Mono;
+
+@Slf4j
+@Component
+public class GpcWebClientFilter {
+    public ExchangeFilterFunction errorHandlingFilter() {
+        return ExchangeFilterFunction.ofResponseProcessor(clientResponse -> {
+            clientResponse.statusCode();
+            if (clientResponse.statusCode().is5xxServerError() || clientResponse.statusCode().is4xxClientError()) {
```

Comment:
"clientResponse.statusCode();" and "if" are in the same indentation. There should be an empty line here.